syntax = "proto2";

package im_user;
import "common.proto"; 
//用户类，用户登录，注销等

//客户端拿验证token来登陆AccessServer
message ulogin
{
	required uint32 imid		= 1; //用户UID
	required bytes imToken	= 2;//用户imToken
	optional uint32  clienttype	= 3;//客户端类型 EnumDefinition.E_CLIENT_TYPE
	optional bytes  deviceInfo	= 4;//手机设备信息
	optional uint32  app_version	= 5;//客户端版本号
	optional bytes  clientip	= 6;//客户端的IP地址ipv4/ipv6:192.168.18.80/2001:db8:1a:1:eeb1:d7ff:fe84:1145/64
	optional bytes  serv_nodeIdentify	= 7;//登录的AccessServer节点信息(AccessServer填写)，客户端不需要填写
	optional uint32  push_chat_msg	= 8;//是否推送单聊，群聊消息，0不推送，1或者无此字段后台推送
}

// 登陆回应
message ulogin_ack
{
	optional common.errorinfo error = 1;//错误码以及错误描述信息
	optional uint32 beat_time	= 2; //客户端心跳时间
	optional uint32 login_prompt = 3;//登录提示 0:正常,1:不是常用设备登录,2:IP异地登录异常
}

//异常登陆客户端确认
message ulogin_comfirm
{
	optional bytes  deviceInfo	= 1;//手机设备信息(把当前设置更新为常用设备)
}
//异常登陆确认,服务端回应
message ulogin_comfirm_ack
{
	optional common.errorinfo error = 1;//错误码以及错误描述信息
}
/////////////////////////////////////////////////////////////////////////////////////////////

//客户端退出IM
message ulogout
{
	required uint32 imid = 1; //用户UID
}


//客户端退出IM响应
message ulogout_ack
{
	optional common.errorinfo error = 1;//错误码以及错误描述信息
}

/////////////////////////////////////////////////////////////////////////////////////////////

//被迫下线通知
message forced_offline_ntf
{
	optional common.errorinfo error = 1;//错误码以及错误描述信息
}





